ui/src/containers/HeaderContainer.js

```diff
@@ -19,30 +24,36 @@ const HeaderContainer = ({ currentUser, history }) => {
   const logout = () => {
     window.location.replace('/oauth/sign_in');
   };
-
-  return (
-    <Header
-      currentUser={currentUser}
-      appName={config.app.name.toUpperCase()}
-      isDropDownOpen={isDropDownOpen}
-      onUserDropdownToggle={onUserDropdownToggle}
-      onTitleClick={onTitleClick}
-      logout={logout}
-    />
-  );
+  if (isUserRequestFailed) {
+    logout();
```

Comment:
I think this might solve the issue we saw on MDC where the screen loads with no data if the URL is manipulated after logout. wdyt @davidffrench @wei-lee

Reply:
Yes it likely would but there are side effects. The user could be legitimately logged in but this specific request to the server fails for whatever reason, we will then be logging the user out.

Reply:
Yes but if requests to the api are failing then the application wont work, and I would consider that a bug. I agree there would be no feedback to the user as to why they are getting logged out. Is that something we should put in the logs of the container so it could be debugged?

Reply:
We do get a 401 response back from this endpoint on authentication failure. With this solution, we should be specifically checking the HTTP response status code for all requests and if it is a 401, display a toast to the user with a message and link advising them to login again.

This will also take into account the token expiring if the page is left open for a long period of time and does not automatically redirect them without them necessarily knowing why.

Reply:
@davidffrench you make a good point, it does sound like a separate piece of work that we could discuss - handling non 200 response codes - we could discuss it at refinement tomorrow?

Reply:
after some discussion, removing this check for now. There is a seperate task to deal with NotFounds here: https://issues.jboss.org/browse/AEROGEAR-9049 so resolving this comment.